flink-runtime/src/main/java/org/apache/flink/runtime/scheduler/adaptive/Executing.java

```diff
@@ -210,6 +265,15 @@ interface Context
          * @return a ScheduledFuture representing pending completion of the task
          */
         ScheduledFuture<?> runIfState(State expectedState, Runnable action, Duration delay);
+
+        /**
+         * Runs the given action immediately or after a delay depending on the given condition.
+         *
+         * @param condition if met, the action is executed immediately or scheduled otherwise
+         * @param action action to run
+         * @param delay delay after which to run the action if the condition is not met
+         */
+        void runIfConditionOrSchedule(boolean condition, Runnable action, Duration delay);
```

Comment:
This interface isn't necessary, if `timeSinceLastRescale().compareTo(scalingIntervalMin)` is true, we can call  `maybeRescale` directly.

If it's false, it's better to call `runIfState` to schedule. 

BTW, if the state isn't this(Executing), the `maybeRescale` shouldn't be called. That's why the `runIfState` name includes `IfState`. The `runIfConditionOrSchedule`  doesn't consider the `IfState`, it might have bug.

Reply:
Agree, thanks